lib/routes/user/create_invoice/create_invoice_page.dart

```diff
@@ -0,0 +1,226 @@
+import 'dart:async';
+import 'package:breez/bloc/account/account_bloc.dart';
+import 'package:breez/bloc/account/account_model.dart';
+import 'package:breez/bloc/app_blocs.dart';
+import 'package:breez/bloc/bloc_widget_connector.dart';
+import 'package:breez/bloc/user_profile/currency.dart';
+import 'package:breez/widgets/static_loader.dart';
+import 'package:flutter/material.dart';
+import 'package:breez/theme_data.dart' as theme;
+import 'package:breez/widgets/back_button.dart' as backBtn;
+import 'package:breez/widgets/amount_form_field.dart';
+import 'package:breez/bloc/invoice/invoice_bloc.dart';
+import 'package:breez/bloc/invoice/invoice_model.dart';
+import 'package:qr_flutter/qr_flutter.dart';
+import 'package:share/share.dart';
+
+class CreateInvoicePage extends StatelessWidget {
+  CreateInvoicePage();
+
+  @override
+  Widget build(BuildContext context) {
+    return new BlocConnector<AppBlocs>((context, blocs) => _CreateInvoicePage(
+        blocs.accountBloc, blocs.invoicesBloc));
+  }
+}
+
+class _CreateInvoicePage extends StatefulWidget {
+  final AccountBloc _accountBloc;
+  final InvoiceBloc _invoiceBloc;
+
+  const _CreateInvoicePage(
+      this._accountBloc, this._invoiceBloc);
+
+  @override
+  State<StatefulWidget> createState() {
+    return new _CreateInvoiceState();
+  }
+}
+
+class _CreateInvoiceState extends State<_CreateInvoicePage> {
+  final _formKey = GlobalKey<FormState>();
+  String _errorMessage = "";
+  String _bolt11 = "";
```

Comment:
I think this widget can be stateless as we don't need all the state variables:
currency can be taken from AccountStream that already listened to in the build function
_bolt11 can also be taken using StreamBuilder in for the readyInvoiceStream in the build function.
This way we can also save the subscriptions.
In general subscriptions in state are only needed when we need to do action that render elements like dialogs and popups that can't be expressed as part of the build function and stateless widget should only be preferred.